code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -90,8 +89,12 @@
 
 	return 1
 /obj/item/weapon/card/emag/attackby(obj/item/O as obj, mob/user as mob)
-	if(istype(O, /obj/item/device/telecrystal))
-		src.uses += default_uses/2 //Adds half the default amount of uses which is more than you get per TC when buying, as to balance the utility of having multiple emags vs one heavily usable one
+	if(istype(O, /obj/item/stack/telecrystal))
+		var/obj/item/stack/telecrystal/T = O
+		if(T.amount/2 < 0)
+			usr << "<span class='notice'>You are not adding enough telecrystals to fuel \the [src].</span>"
+			return
+		src.uses += T.amount/2 //Gives 5 uses per 10 TC

```

Comment:
I actually have no idea. I suppose I assumed that it would be handled by whatever.